### ISSUE TITLE
Error adding access keypad

### ISSUE THREAD

**Issue by ntlgr** [rospogrigio/localtuya#1546]
## The problem
I have a Tuya Access Keypad and I cannot add it to localtuya.
The device can be seen [here](https://www.aliexpress.com/item/4000334211110.html). 

## Environment

- Localtuya version:  v5.2.1 from HACS
- Home Assistant Core version: Home Assistant 2023.10.3 Supervisor 2023.10.0 Operating System 11.0 Frontend 20231005.0 - latest
- [x] Does the device work using the Home Assistant Tuya Cloud component ?  yes
- [] Does the device work using the Tinytuya (https://github.com/jasonacox/tinytuya) command line tool ?  not tested
- [] Was the device working with earlier versions of localtuya ? Which one?  not tested
- [x] Are you using the Tuya/SmartLife App in parallel ?  yes

## Steps to reproduce

1. go to devices -> local tuya integration -> configure
2. add new device and select device from the list
3. fill in details and local key, hit submit
4. Error occurs: "Connection to device succeeded but no datapoints found, please try again. Create a new issue and include debug logs if problem persists."
This error occurs with protocols 3.1, 3.2, 3.3. With protocol 3.4 it just times out after a few minutes.

## DP dump

Unable to run DP dump. The local key of the device ends with a ")" and the script fails.
I have examined the device status and found out what changes when it sends unlock signal. See below.

      {
        "code": "closed_opened_kit",
        "custom_name": "",
        "dp_id": 35,
        "time": 1694074129492,
        "value": "AQAC"  _**---------> this becomes AQAB when the door unlocks**_
      },

## Provide Home Assistant traceback/logs

```
023-10-26 10:50:16.612 DEBUG (MainThread) [custom_components.localtuya.pytuya] [bfe...kwk] Sending command 10 (device type: type_0a)
2023-10-26 10:50:16.612 DEBUG (MainThread) [custom_components.localtuya.pytuya] [bfe...kwk] Sending payload: b'{"gwId":"bfe3c7b09272296fechkwk","devId":"bfe3c7b09272296fechkwk","uid":"bfe3c7b09272296fechkwk","t":"1698306616"}'
2023-10-26 10:50:16.616 DEBUG (MainThread) [custom_components.localtuya.pytuya] [bfe...kwk] Command 10 waiting for seq. number 1
2023-10-26 10:50:16.623 DEBUG (MainThread) [custom_components.localtuya.pytuya] [bfe...kwk] Dispatching message CMD 10 TuyaMessage(seqno=1, cmd=10, retcode=1, payload=b'\xc3\x051\x04\xab\x11\x00\xca:\x1a\x9d\xa9\r\xf1~\x86\xf1\x87\xbf\xca\x07v\xa3\xf5j\x94\x063\xb3\x87\x8dM', crc=1863176706, crc_good=True)
2023-10-26 10:50:16.624 DEBUG (MainThread) [custom_components.localtuya.pytuya] [bfe...kwk] 'data unvalid' error detected: switching to dev_type 'type_0d'
2023-10-26 10:50:16.624 DEBUG (MainThread) [custom_components.localtuya.pytuya] [bfe...kwk] Re-send 10 due to device type change (type_0a -> type_0d)
2023-10-26 10:50:16.624 DEBUG (MainThread) [custom_components.localtuya.pytuya] [bfe...kwk] Sending command 10 (device type: type_0d)
2023-10-26 10:50:16.625 DEBUG (MainThread) [custom_components.localtuya.pytuya] [bfe...kwk] Sending payload: b'{"devId":"bfe3c7b09272296fechkwk","uid":"bfe3c7b09272296fechkwk","t":"1698306616","dps":{"1":null,"2":null,"3":null,"4":null,"5":null,"6":null,"7":null,"8":null,"9":null,"10":null}}'
2023-10-26 10:50:16.627 DEBUG (MainThread) [custom_components.localtuya.pytuya] [bfe...kwk] Command 13 waiting for seq. number 2
2023-10-26 10:50:16.697 DEBUG (MainThread) [custom_components.localtuya.pytuya] [bfe...kwk] Dispatching message CMD 8 TuyaMessage(seqno=0, cmd=8, retcode=0, payload=b'3.3\x00\x00\x00\x00\x00\x00\x99\xf6\x00\x00\x00\x01\xa4MF\xacc\x04b(y\x0c\xdb\x1a\xbb\xd9\x80 \t\x9aB\xb0Uu\x88!\x87f\x8fo.\xf1\x14\xb7z\xf8\xac~\xb6_Q\x90\xfe\x9c\xd8\xabV\\\x05\x94', crc=1698055876, crc_good=True)
2023-10-26 10:50:16.697 DEBUG (MainThread) [custom_components.localtuya.pytuya] [bfe...kwk] Got status update
2023-10-26 10:50:16.700 DEBUG (MainThread) [custom_components.localtuya.pytuya] [bfe...kwk] Deciphered data = '{"dps":{},"type":"query","t":1698306615}'
2023-10-26 10:50:16.739 DEBUG (MainThread) [custom_components.localtuya.pytuya] [bfe...kwk] Dispatching message CMD 13 TuyaMessage(seqno=2, cmd=13, retcode=0, payload=b'', crc=2380353348, crc_good=True)
2023-10-26 10:50:16.740 DEBUG (MainThread) [custom_components.localtuya.pytuya] [bfe...kwk] ACK received for command 13: ignoring it
2023-10-26 10:50:16.740 DEBUG (MainThread) [custom_components.localtuya.pytuya] [bfe...kwk] Sending command 10 (device type: type_0d)
2023-10-26 10:50:16.740 DEBUG (MainThread) [custom_components.localtuya.pytuya] [bfe...kwk] Sending payload: b'{"devId":"bfe3c7b09272296fechkwk","uid":"bfe3c7b09272296fechkwk","t":"1698306616","dps":{"1":null,"11":null,"12":null,"13":null,"14":null,"15":null,"16":null,"17":null,"18":null,"19":null,"20":null}}'
2023-10-26 10:50:16.744 DEBUG (MainThread) [custom_components.localtuya.pytuya] [bfe...kwk] Command 13 waiting for seq. number 3
2023-10-26 10:50:16.821 DEBUG (MainThread) [custom_components.localtuya.pytuya] [bfe...kwk] Dispatching message CMD 8 TuyaMessage(seqno=0, cmd=8, retcode=0, payload=b'3.3\x00\x00\x00\x00\x00\x00\x99\xf7\x00\x00\x00\x01\xa4MF\xacc\x04b(y\x0c\xdb\x1a\xbb\xd9\x80 \t\x9aB\xb0Uu\x88!\x87f\x8fo.\xf1\x14\xb7z\xf8\xac~\xb6_Q\x90\xfe\x9c\xd8\xabV\\\x05\x94', crc=468645509, crc_good=True)
2023-10-26 10:50:16.821 DEBUG (MainThread) [custom_components.localtuya.pytuya] [bfe...kwk] Got status update
2023-10-26 10:50:16.821 DEBUG (MainThread) [custom_components.localtuya.pytuya] [bfe...kwk] Deciphered data = '{"dps":{},"type":"query","t":1698306615}'
2023-10-26 10:50:16.864 DEBUG (MainThread) [custom_components.localtuya.pytuya] [bfe...kwk] Dispatching message CMD 13 TuyaMessage(seqno=3, cmd=13, retcode=0, payload=b'', crc=1350014657, crc_good=True)
2023-10-26 10:50:16.864 DEBUG (MainThread) [custom_components.localtuya.pytuya] [bfe...kwk] ACK received for command 13: ignoring it
2023-10-26 10:50:16.864 DEBUG (MainThread) [custom_components.localtuya.pytuya] [bfe...kwk] Sending command 10 (device type: type_0d)
2023-10-26 10:50:16.865 DEBUG (MainThread) [custom_components.localtuya.pytuya] [bfe...kwk] Sending payload: b'{"devId":"bfe3c7b09272296fechkwk","uid":"bfe3c7b09272296fechkwk","t":"1698306616","dps":{"1":null,"21":null,"22":null,"23":null,"24":null,"25":null,"26":null,"27":null,"28":null,"29":null,"30":null}}'
2023-10-26 10:50:16.868 DEBUG (MainThread) [custom_components.localtuya.pytuya] [bfe...kwk] Command 13 waiting for seq. number 4
2023-10-26 10:50:16.944 DEBUG (MainThread) [custom_components.localtuya.pytuya] [bfe...kwk] Dispatching message CMD 8 TuyaMessage(seqno=0, cmd=8, retcode=0, payload=b'3.3\x00\x00\x00\x00\x00\x00\x99\xf8\x00\x00\x00\x01\xa4MF\xacc\x04b(y\x0c\xdb\x1a\xbb\xd9\x80 \t\x9aB\xb0Uu\x88!\x87f\x8fo.\xf1\x14\xb7\xb8&\x15\x13\xb7\x8f\x0eA\xfb\xbe\x8e\x10\xdeI{(', crc=3693166571, crc_good=True)
2023-10-26 10:50:16.945 DEBUG (MainThread) [custom_components.localtuya.pytuya] [bfe...kwk] Got status update
2023-10-26 10:50:16.945 DEBUG (MainThread) [custom_components.localtuya.pytuya] [bfe...kwk] Deciphered data = '{"dps":{},"type":"query","t":1698306616}'
2023-10-26 10:50:16.997 DEBUG (MainThread) [custom_components.localtuya.pytuya] [bfe...kwk] Dispatching message CMD 13 TuyaMessage(seqno=4, cmd=13, retcode=0, payload=b'', crc=755273881, crc_good=True)
2023-10-26 10:50:16.997 DEBUG (MainThread) [custom_components.localtuya.pytuya] [bfe...kwk] ACK received for command 13: ignoring it
2023-10-26 10:50:16.997 DEBUG (MainThread) [custom_components.localtuya.pytuya] [bfe...kwk] Sending command 10 (device type: type_0d)
2023-10-26 10:50:16.997 DEBUG (MainThread) [custom_components.localtuya.pytuya] [bfe...kwk] Sending payload: b'{"devId":"bfe3c7b09272296fechkwk","uid":"bfe3c7b09272296fechkwk","t":"1698306616","dps":{"1":null,"100":null,"101":null,"102":null,"103":null,"104":null,"105":null,"106":null,"107":null,"108":null,"109":null,"110":null}}'
2023-10-26 10:50:16.998 DEBUG (MainThread) [custom_components.localtuya.pytuya] [bfe...kwk] Command 13 waiting for seq. number 5
2023-10-26 10:50:17.083 DEBUG (MainThread) [custom_components.localtuya.pytuya] [bfe...kwk] Dispatching message CMD 8 TuyaMessage(seqno=0, cmd=8, retcode=0, payload=b'3.3\x00\x00\x00\x00\x00\x00\x99\xf9\x00\x00\x00\x01\xa4MF\xacc\x04b(y\x0c\xdb\x1a\xbb\xd9\x80 \t\x9aB\xb0Uu\x88!\x87f\x8fo.\xf1\x14\xb7\xb8&\x15\x13\xb7\x8f\x0eA\xfb\xbe\x8e\x10\xdeI{(', crc=2734264234, crc_good=True)
2023-10-26 10:50:17.083 DEBUG (MainThread) [custom_components.localtuya.pytuya] [bfe...kwk] Got status update
2023-10-26 10:50:17.084 DEBUG (MainThread) [custom_components.localtuya.pytuya] [bfe...kwk] Deciphered data = '{"dps":{},"type":"query","t":1698306616}'
2023-10-26 10:50:17.125 DEBUG (MainThread) [custom_components.localtuya.pytuya] [bfe...kwk] Dispatching message CMD 13 TuyaMessage(seqno=5, cmd=13, retcode=0, payload=b'', crc=4036118812, crc_good=True)
2023-10-26 10:50:17.126 DEBUG (MainThread) [custom_components.localtuya.pytuya] [bfe...kwk] ACK received for command 13: ignoring it
2023-10-26 10:50:17.126 DEBUG (MainThread) [custom_components.localtuya.pytuya] [bfe...kwk] Detected dps: {}
2023-10-26 10:50:17.126 DEBUG (MainThread) [custom_components.localtuya.config_flow] Detected DPS: {}
2023-10-26 10:50:17.126 DEBUG (MainThread) [custom_components.localtuya.pytuya] [bfe...kwk] Closing connection
2023-10-26 10:50:17.132 DEBUG (MainThread) [custom_components.localtuya.pytuya] [bfe...kwk] Connection lost: None
```
